### PR TITLE
feat: remember last logged in username [DET-3274]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

the last username used to log in with a Determined (non SSO) account would prefill the username field for subsequent visits to the login form.

## Commentary (optional)

Clearing the browser/site data would remove the "last username used"

This means that if a user logs once with a determined username "x" and logs in using SSO every subsequent time, they will still see the "X" show up in the username field every time.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/components/DeterminedAuth.tsx [hunk 1]
```diff
@@ -16,6 +17,9 @@ interface FromValues {
   username?: string;
 }
 
+const storage = new Storage({ basePath: '/DeterminedAuth', store: window.localStorage });
+const LAST_USERNAME = 'lastUsername'; // key for storing last det username
```
Comment (hkang1) [inline webui/react/src/components/DeterminedAuth.tsx:21]: suggestion: naming the constant to `STORAGE_KEY_USERNAME` and dropping the comment

File: webui/react/src/components/DeterminedAuth.tsx [hunk 2]
```diff
@@ -29,11 +33,13 @@ const DeterminedAuth: React.FC = () => {
       await login(creds as Credentials);
       const user = await getCurrentUser({});
       setAuth({ type: Auth.ActionType.Set, value: { isAuthenticated: true, user } });
+      storage.set(LAST_USERNAME, creds.username);
     } catch (e) {
       const isBadCredentialsSync = isLoginFailure(e);
       setIsBadCredentials(isBadCredentialsSync); // this is not a sync operation
       setShowSpinner({ type: FullPageSpinner.ActionType.Hide });
       const actionMsg = isBadCredentialsSync ? 'check your username and password.' : 'retry.';
+      if (isBadCredentialsSync) storage.remove(LAST_USERNAME); // forget saved username on first bad auth
```
Comment (hkang1) [inline webui/react/src/components/DeterminedAuth.tsx:42]: nit: IMO, I think this line reads pretty nicely where the comment is not needed

File: webui/react/src/components/DeterminedAuth.tsx [hunk 3]
```diff
@@ -16,6 +17,9 @@ interface FromValues {
   username?: string;
 }
 
+const storage = new Storage({ basePath: '/DeterminedAuth', store: window.localStorage });
```
Comment (hkang1) [inline webui/react/src/components/DeterminedAuth.tsx:20]: question: How about using the `useStorage` hook? Will it not work because of the user namespace?

Comment (hamidzr) [reply webui/react/src/components/DeterminedAuth.tsx:20]: yeah I think the login changes the namespace for that

**Pull request merged**